content/oci/oci.go

```diff
@@ -155,7 +201,7 @@ func (s *Store) tag(ctx context.Context, desc ocispec.Descriptor, reference stri
 		return err
 	}
 	if s.AutoSaveIndex {
-		return s.SaveIndex()
+		return s.saveIndex()
```

Comment:
Why `Tag()` can change `index.json` but only get read Lock but not write Lock?

Reply:
We use read locks for all operations other than delete. We do this because we have optimized the underlying implementation to be concurrently safe.